desktop/platform/index.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+/**
+ * Internal dependencies
+ */
+const os = require( 'os' );
+
+/**
+ * Module variables
+ */
+let platform = false;
+
+function Platform() {
+	this.platform = false;
+}
+
+Platform.prototype.isOSX = function() {
+	return process.platform === 'darwin';
+};
+
+Platform.prototype.isWindows = function() {
+	return process.platform === 'win32';
+};
+
+Platform.prototype.isLinux = function() {
+	return process.platform === 'linux';
+};
+
+if ( ! platform ) {
+	platform = new Platform();
+}
+
+module.exports = platform;
```

Comment:
A little complicated?

```js
const os = require( 'os' );

const platformInfo = ( { platform } ) => ( {
	isOSX: platform === 'darwin',
	isWindows: platform === 'win32',
	isLinux: platform === 'linux'
} );

module.exports = platformInfo( process );
```

seems simpler _and_ it's testable if we were to export the `platformInfo` function.